### PR TITLE
chore(deps): bump svelty-picker from 3.6.1 to 5.0.0

### PR DESCRIPTION
**Pull request by VmMad**
## Summary

> bump svelty-picker version and cleanup DateTimePicker

## Changelog

```
- Bump svelty-picker version to latest
- Remove unused css vars
- Cleanup DateTimePicker types and code composition
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows

### Instructions

...

## Checklist

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/inputs/DateTimePicker.svelte [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
+    import { ComponentProps, createEventDispatcher } from 'svelte'
     import { Tooltip, Button, ButtonSize } from 'shared/components'
     import { localize } from '@core/i18n'
-    import sveltyPickerTranslations from 'svelty-picker/i18n'
     import { appSettings } from '@core/app'
 
-    export let value: Date = undefined
-    export let startDate: Date = null
+    export let value: Date | undefined = undefined
+    export let startDate: Date | null = null
```
Comment (begonaalvarezd) [inline packages/shared/components/inputs/DateTimePicker.svelte:11]: couldnt we unify these to be undefined or null both?

File: packages/shared/components/inputs/DateTimePicker.svelte [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
     export let mode: 'auto' | 'datetime' | 'date' | 'time' = 'auto'
 
     const dispatch = createEventDispatcher()
     const sveltyPickerStartDate = convertDateToSveltyPickerFormat(startDate)
-
-    let sveltyPickerDate = convertDateToSveltyPickerFormat(value) ?? sveltyPickerStartDate
     const translations = getSveltyPickerTranslations()
+
+    let sveltyPickerDate = value ? convertDateToSveltyPickerFormat(value) : sveltyPickerStartDate
```
Comment (begonaalvarezd) [inline packages/shared/components/inputs/DateTimePicker.svelte:18]: with null/undefined types unified you can simplify this too and leave it as it was before

File: packages/shared/components/inputs/DateTimePicker.svelte [hunk 3]
```diff
@@ -1,41 +1,43 @@
...
     let tooltip: Tooltip
 
-    function convertDateToSveltyPickerFormat(date: Date): string {
-        return date?.toLocaleString('sv')
+    function convertDateToSveltyPickerFormat(date: Date | null): string | undefined {
+        return date ? date?.toLocaleString('sv') : undefined
```
Comment (begonaalvarezd) [inline packages/shared/components/inputs/DateTimePicker.svelte:22]: same here, no need to `date ? date?.`, eg
```suggestion
    function convertDateToSveltyPickerFormat(date: Date | undefined): string | undefined {
        return date?.toLocaleString('sv') ?? undefined
```

File: packages/shared/components/inputs/DateTimePicker.svelte [hunk 4]
```diff
@@ -1,41 +1,43 @@
...
     }
 
     function onCancelClick(): void {
         dispatch('cancel')
     }
 
     function onConfirmClick(): void {
-        value = new Date(sveltyPickerDate)
-        dispatch('confirm')
+        if (sveltyPickerDate) {
```
Comment (begonaalvarezd) [inline packages/shared/components/inputs/DateTimePicker.svelte:30]: if there is no date delected you can press confirm or the action will do nothing, I guess we should allow proceeding even if here is no date selected, you can set nothing

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Very good work, added some comments
Additionally, I cant install the app with the changes, it errors 🙏🏼

**Review by cpl121 — CHANGES_REQUESTED** [review]
image

The current date does not appear marked when I open the popup
Also, I have the application in English and the date appears in Spanish

**Comment by VmMad** [comment]
> image
> 
> The current date does not appear marked when I open the popup Also, I have the application in English and the date appears in Spanish

@cpl121 did you run `yarn install` before testing the PR? because I upgraded the version and the style variables changed and you probably didnt sync the package. For me the selected day shows up, and the problem with the spanish picker is the reactivity problem with i18n, because I didn't change anything and for me it shows in english. Try starting the dev server from zero. `yarn && cd packages/desktop && yarn && yarn build && yarn start`

Screenshot from 2023-10-18 10-45-37

**Comment by cpl121** [comment]
> > image
> > The current date does not appear marked when I open the popup Also, I have the application in English and the date appears in Spanish
> 
> @cpl121 did you run `yarn install` before testing the PR? because I upgraded the version and the style variables changed and you probably didnt sync the package. For me the selected day shows up, and the problem with the spanish picker is the reactivity problem with i18n, because I didn't change anything and for me it shows in english. Try starting the dev server from zero. `yarn && cd packages/desktop && yarn && yarn build && yarn start`
> 
> Screenshot from 2023-10-18 10-45-37

Yes, I have run the large command 🥲

**Review by cpl121 — APPROVED** [review]
Perfect!! 🔥

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**